### PR TITLE
add sparse index option

### PR DESCRIPTION
**Pull request by nkosmol-braintags**

### CODE REVIEW

File: vertx-pojo-mapper-common/pom.xml [hunk 1]
```diff
@@ -25,4 +25,11 @@
 	<description>common interfaces and implementations for pojo mapper
 	</description>
 
+	<dependencies>
+		<dependency>
+			<groupId>commons-codec</groupId>
+			<artifactId>commons-codec</artifactId>
+			<version>1.4</version>
```
Comment (sschmittBt) [inline vertx-pojo-mapper-common/pom.xml:32]: vertx-util already has this dependency in its gradle file, and with version 1.10

**Review by sschmittBt — CHANGES_REQUESTED** [review]

File: vertx-pojo-mapper-common/pom.xml (hunk 1 above)
Comment (remi128) [reply vertx-pojo-mapper-common/pom.xml:32]: but this is not helping in maven

File: vertx-pojongo/src/main/java/de/braintags/vertx/jomnigate/mongo/MongoUtil.java [hunk 2]
```diff
@@ -210,22 +211,27 @@ private static void addIndexOptions(final JsonObject indexDef,
...
+        default:
+          future.fail(new IllegalArgumentException("Unknown IndexFeature: " + option.getFeature()));
       }
     }
     CompositeFuture.all(futures).setHandler(result -> handler.handle(result.map(indexDef)));
   }
 
-  private static void convertFilterExpression(String filterExpression, final JsonObject indexDef, IMapper<?> mapper,
-      final MongoDataStore dataStore, Future<Void> future) {
+  private static void convertFilterExpression(final String filterExpression, final JsonObject indexDef,
```
Comment (remi128) [inline vertx-pojongo/src/main/java/de/braintags/vertx/jomnigate/mongo/MongoUtil.java:231]: please check your formatting options, auto format, cause linebreaks should be different here

**Review by remi128 — CHANGES_REQUESTED** [review]

File: vertx-pojo-mapper-common/pom.xml (hunk 1 above)
Comment (sschmittBt) [reply vertx-pojo-mapper-common/pom.xml:32]: then it should be added in vertx-util for maven, but not here

**Review by remi128 — APPROVED** [review]

**Review by sschmittBt — APPROVED** [review]

**Pull request merged**